### PR TITLE
Change issue 1945 test to paramaterized

### PR DESCRIPTION
**Pull request by ghost**
Hello, I have worked with @TheJosephCCox to address the code smell concern raised in #3668. Please let us know what you think of this change. Thank you.

### CODE REVIEW

File: javaparser-symbol-solver-testing/src/test/java/com/github/javaparser/symbolsolver/Issue1945Test.java [hunk 1]
```diff
@@ -17,69 +17,78 @@
...
 
-        CompilationUnit cu = StaticJavaParser.parse(code);
-        List<MethodCallExpr> nodes = cu.findAll(MethodCallExpr.class);
-        for (MethodCallExpr expr : nodes) {
-            String qName = expr.resolve().getQualifiedName();
-            String resolvedType = expr.calculateResolvedType().describe();
-            assertEquals(resultsQualifiedName.get(expr.toString()), qName);
-            assertEquals(resultsResolvedType.get(expr.toString()), resolvedType);
-        }
+    private static List<MethodCallExpr> parsedCodeMethodCalls() {
```
Comment (jlerbsc) [inline javaparser-symbol-solver-testing/src/test/java/com/github/javaparser/symbolsolver/Issue1945Test.java:73]: Why are you declaring this method static?

File: javaparser-symbol-solver-testing/src/test/java/com/github/javaparser/symbolsolver/Issue1945Test.java [hunk 2]
```diff
@@ -17,69 +17,78 @@
...
+        StaticJavaParser.setConfiguration(config);
         
+        CompilationUnit cu = StaticJavaParser.parse(code);
+
+        return cu.findAll(MethodCallExpr.class);
     }
 
+    @ParameterizedTest
+    @MethodSource("parsedCodeMethodCalls")
+    void test(MethodCallExpr expr) {
```
Comment (jlerbsc) [inline javaparser-symbol-solver-testing/src/test/java/com/github/javaparser/symbolsolver/Issue1945Test.java:88]: Indeed this test method is a little less complex in this case.

File: javaparser-symbol-solver-testing/src/test/java/com/github/javaparser/symbolsolver/Issue1945Test.java (hunk 1 above)
Comment (TheJosephCCox) [reply javaparser-symbol-solver-testing/src/test/java/com/github/javaparser/symbolsolver/Issue1945Test.java:73]: It is our understanding that the factory method that provides inputs to a test must be declared static as per the [JUnit documentation](https://junit.org/junit5/docs/current/user-guide/#writing-tests-parameterized-tests-sources-MethodSource).

**Comment by jlerbsc** [comment]
Thank you for this contribution.

**Pull request merged**